sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -532,8 +532,8 @@ private static bool IsFinalState(Response response, HeaderSource headerSource, o
                         switch (headerSource)
                         {
                             case HeaderSource.None when root.TryGetProperty("properties", out var properties) && properties.TryGetProperty("provisioningState", out JsonElement property):
-                            case HeaderSource.OperationLocation when root.TryGetProperty("status", out property):
-                            case HeaderSource.AzureAsyncOperation when root.TryGetProperty("status", out property):
+                            case HeaderSource.OperationLocation when root.TryGetProperty("properties", out properties) && properties.TryGetProperty("provisioningState", out property):
```

Comment:
This needs to be reverted.   You should not be making changes to Azure Core files here.

Reply:
Reverted the changes made in the core files